### ISSUE TITLE
Can't prevent zip package builds, slows compile times

### ISSUE THREAD

**Issue by profexorgeek** [rlabrecque/Steamworks.NET#531]
My game is built with FlatRedBall, which is MonoGame under the hood. I'm using the Standalone from release v20.1 and have successfully implemented achievements and stats via this library (thank you!).

The problem is that I use the Steam Builder to directly push files to Steam. But every time I build hundreds of megabytes of zip files are being created, vastly increasing my compile time:
![image](https://user-images.githubusercontent.com/711100/204099279-52fe9037-f7c8-4d4a-aa52-e78e41d5cb47.png)

This would be fine for RELEASE but in DEBUG I don't want to waste the time creating all of these zip files. However, I can't seem to figure out how to turn this option off.

Is there a way to disable zip package creation when using the standalone DLLs?

**Comment by profexorgeek** [comment]
In case it matters, my project Target Framework is .NET Framework 4.7.2

**Comment by rlabrecque** [comment]
Hmm, the Steam Builder or FlatRedBall repos might be a better place for this? Steamworks (and Steamworks.NET by extension) doesn't care or know about zips at all in general.

**Comment by profexorgeek** [comment]
I’m a primary contributor to the Flat Red Ball game engine so I’m reasonably certain it’s not that.

The Steam Builder is a standalone CLI application and not part of the visual studio build pipeline.

My game has been released for years and this build process change just started when I added achievements via Steamworks.NET last month.

So there’s nothing in Steamworks that emits zip packages during the Build process?

**Comment by rlabrecque** [comment]
Nope, there isn't even really a build process. Steamworks is just a collection of runtime APIs in some dlls that you need to link to.

Steamworks.NET has a build process to produce the "Standalone" dlls, but all that does is does is produces a dll.

Do you have a link for Steam Builder?

**Comment by profexorgeek** [comment]
Okay, first of all my initial post had some confusing things in it. I did a bunch of research and it looks like you're right - there's nothing in SteamWorks.NET or SteamWorks itself that should affect Visual Studio builds according to any docs I've found. It's really strange that it started happening when I added the steamAPI and SteamWorks.NET DLLs. Now I'm wondering if there was some csproj change that affected the build system. I will review my git log.

To answer your questions: "SteamBuilder" is apparently not what it's called, that's just what I have called it for years. I was actually referring to the SteamPipe as documented here: https://partner.steamgames.com/doc/sdk/uploading

My release process is:

- Set VS to RELEASE
- Build the game
- Copy built files from the `bin` directory into a content folder in the SteamPipe directory
- Run steam pipe batch files to push built content to Steam
- Test in a beta branch, then set the build live to the main branch

So SteamPipe is in no way integrated with Visual Studio. When I'm not building for RELEASE, I just F5 to build in DEBUG mode and quickly test my game. It used to take it seconds to compile, now it takes minutes. My game, Masteroid, has been out for years but I decided to add Achievements due to community requests. I shipped that feature on October 27:
https://steamcommunity.com/app/1148320

It was during that update that my build system suddenly started emitting zip files of the game for every platform. So, I assumed that the DLL was somehow injecting itself into the process to build zip files for each platform. Especially since these zip files seem specifically intended to upload to Steam!

If it's okay with you, I'll leave this issue open and close it when I find a solution?

**Comment by profexorgeek** [comment]
Okay, this is embarrassing. When I dusted off my codebase in October, I had to upgrade a ton of stuff to make the game build again after years of not touching it.

Apparently I had a Nuget package from years ago that I forgot about and wasn't using. I updated it as part of bringing the code up to speed. The package's whole purpose is to publish zip files that can be uploaded to Steam (the old version required CLI to create the packages). If anyone cares, it's here:
https://www.nuget.org/packages/InfinitespaceStudios.MonoGame.Packaging/0.1.10-alpha

I'm sorry to have wasted your time. That nuget update was lost in a commit full of MonoGame and csproj cleanup and I missed it the first time I looked... 🤦‍♂️
